### PR TITLE
Zendesk: show alert if user email is a8c

### LINKED ISSUE #11024: Zendesk Mobile: cannot use "Contact Us" if the email in Zendesk matches the email used to log in to WPiOS via WordPress.com
### Expected behavior

I expect to be able to send a message through Me > Help & Support > Contact Us regardless of what email address is used.

### Actual behavior

If the email address used to log in to WPiOS via WordPress.com is also used for a Zendesk account, you cannot send a message.

### Steps to reproduce the behavior

1. Log in to WPiOS via WordPress.com with an account that uses the same email address that you also use for a Zendesk account.
1. Go to Me > Help & Support.
1. Set a "Contact Email" if you haven't already.
1. Go to Contact Us.
1. Try sending a message. 

Result: message does not send and the following error appears:

> Message failed to send. Tap to retry

##### Tested on iPhone 6S, iOS 12.1.4, WP Internal 11.8.0.20190211.

Note: @ScoutHarris mentioned that this is a Zendesk limitation that we cannot fix and that the solution will be to show an alert.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #11024 

Zendesk does not allow agents to submit tickets, and displays a `Message failed to send` error upon attempt. We can't do anything about that, so this change shows a warning when an `@automattic.com` or `@a8c.com` email address is detected:
- When the Support view is initially shown.
- When the user changes their email via `Contact Email`.

To test:
- Go to Me > Help & Support.
- If your email address is `@automattic.com`/`@a8c.com`, an alert should immediately appear.
- Select `Contact Email` and change your email to `@automattic.com`/`@a8c.com`. When `OK`, an alert should appear.

warning

To note, you can still attempt to create a ticket using an a8c email address. It will still fail with the`Message failed to send` error. But, you've been warned. 😉

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@nheagy - would you mind giving this a go? Thank you!

**Comment by ScoutHarris** [comment]
Hey @nheagy .

> it doesn't block my @a8c.com email address

🤦‍♂️ you'd think I'd have thought to include that. It shall be done! I'll ping you when it's done. Thank you!

**Comment by ScoutHarris** [comment]
Hey @nheagy . Ready for another review please kind sir. The code and PR description have been updated to include checking for `@a8c.com`.

File: WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift [hunk 1]
```diff
@@ -220,10 +221,29 @@ private extension SupportTableViewController {
...
             }
         }
     }
 
+    /// Zendesk does not allow agents to submit tickets, and displays a 'Message failed to send' error upon attempt.
+    /// If the user email address is a8c, display a warning.
+    ///
+    func checkForAutomatticEmail() {
+        guard let email = ZendeskUtils.userSupportEmail(),
+            (Constants.automatticEmails.first { email.contains($0) == true }) != nil else {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift]: You could drop the `== true` and it will still work, fwiw.

Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift]: Well, you would be correct, and tis been removed. Thank you!

**Pull request merged**